src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -3021,60 +3081,316 @@ partial class C
-                //     public partial string? M8(); // 5
+                //     public partial string? M8();
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(32, 26),
+                // (33,27): warning CS8824: Partial method declarations 'string? C.M9()' and 'string C.M9()' have differences in parameter or return types.
+                //     public partial string M9() => null; // 10
+                Diagnostic(ErrorCode.WRN_PartialMethodTypeDifference, "M9").WithArguments("string? C.M9()", "string C.M9()").WithLocation(33, 27),
                 // (35,26): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
-                //     public partial string? M11() => null; // 6
+                //     public partial string? M11() => null;
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(35, 26),
                 // (36,26): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
-                //     public partial string? M12(); // 7
+                //     public partial string? M12();
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(36, 26));
         }
 
+        [Fact]
+        public void DifferentReturnTypes_18()
+        {
+            var source =
+@"partial class C
+{
+    public partial ref int F1();
+    public partial string F1() => null;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (4,27): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial string F1() => null;
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "F1").WithLocation(4, 27),
+                // (4,27): error CS8818: Partial method declarations must have matching ref return values.
+                //     public partial string F1() => null;
+                Diagnostic(ErrorCode.ERR_PartialMethodRefReturnDifference, "F1").WithLocation(4, 27));
+        }
+
+        [Fact]
+        public void DifferentReturnTypes_19()
+        {
+            var source =
+@"partial class C
+{
+    public partial ref (int x, int y) F1();
+    public partial (int, int) F1() => default;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (4,31): error CS8818: Partial method declarations must have matching ref return values.
+                //     public partial (int, int) F1() => default;
+                Diagnostic(ErrorCode.ERR_PartialMethodRefReturnDifference, "F1").WithLocation(4, 31));
```

Comment:
it feels like it would be good to also give an error for mismatching tuple names here, but don't go too far out of your way to make it so.

Reply:
> it would be good to also give an error for mismatching tuple names here

This behavior is actually unchanged. `MemberSignatureComparer.ConsideringTupleNamesCreatesDifference()` only returns true if the differences are limited to tuple element names.

---
In reply to: [489812714](https://github.com/dotnet/roslyn/pull/47576#discussion_r489812714) [](ancestors = 489812714)